Squashed commit of the following:

ricardoquesada opened the pull request:
commit aff4e27200a77db60b13ea30c2457558e5f53059
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 17:58:41 2015 -0800

```
compiles with new GC model
```

commit 1fa69cd71231d56371cd45a378e50a1888308b42
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 17:41:15 2015 -0800

```
Animation3D works Ok
```

commit d439969caf7e6fe83a74e37d078c4361a08cb816
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:39:50 2015 -0800

```
Sprite3d create: converted to new API
```

commit aabe449e4a968fad882c44df9be787eb7c3bfcec
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:27:33 2015 -0800

```
ouch...
```

commit 688ab610a8cb7607bc3c51b8ca01d800ef3c9794
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:12:25 2015 -0800

```
Spawn/Sequence init* were not bound

Fixed. Now JS Scheduler Test works Ok
```

ricardoquesada committed:
Squashed commit of the following:

commit aff4e27200a77db60b13ea30c2457558e5f53059
Author: Ricardo Quesada <ricardoquesada@gmail.com>
Date:   Thu Dec 10 17:58:41 2015 -0800

    compiles with new GC model

commit 1fa69cd71231d56371cd45a378e50a1888308b42
Author: Ricardo Quesada <ricardoquesada@gmail.com>
Date:   Thu Dec 10 17:41:15 2015 -0800

    Animation3D works Ok

commit d439969caf7e6fe83a74e37d078c4361a08cb816
Author: Ricardo Quesada <ricardoquesada@gmail.com>
Date:   Thu Dec 10 13:39:50 2015 -0800

    Sprite3d create: converted to new API

commit aabe449e4a968fad882c44df9be787eb7c3bfcec
Author: Ricardo Quesada <ricardoquesada@gmail.com>
Date:   Thu Dec 10 13:27:33 2015 -0800

    ouch...

commit 688ab610a8cb7607bc3c51b8ca01d800ef3c9794
Author: Ricardo Quesada <ricardoquesada@gmail.com>
Date:   Thu Dec 10 13:12:25 2015 -0800

    Spawn/Sequence init* were not bound

    Fixed. Now JS Scheduler Test works Ok

ricardoquesada committed:
Adds MemoryModelTest

zilongshanren commented:
@ricardoquesada 
:+1:  Finally the JSB memory model has been improved :smile: 

Should we include this improvement in v3.10 or the future release?
Now we have a separate v3.10 branch in the upstream.

ricardoquesada committed:
Mooooores fixes

but GC memory model doesn't work yet... somewhere someone is still
retaining a reference that prevents the whole thing from
releasing

pandamicro commented on cocos/scripting/js-bindings/manual/ScriptingCore.cpp:
```diff
@@ -430,8 +430,9 @@ void registerDefaultClasses(JSContext* cx, JS::HandleObject global) {
 
-static void sc_finalize(JSFreeOp *freeOp, JSObject *obj) {
-    CCLOGINFO("jsbindings: finalizing JS object %p (global class)", obj);
+static void sc_finalize(JSFreeOp *freeOp, JSObject *obj)
```
What is this function used for ?

ricardoquesada replied:
I don't know... someone is using it. I just did some minor coding-style fixes

pandamicro commented on tests/js-tests/src/MemoryModelTest/MemoryModelTest.js:
```diff
@@ -0,0 +1,256 @@
+        // add random property
+        sprite.randomProperty = "hello world";
+
+        sprite = this.getChildByTag(tag);
```
This should be placed in a scheduled callback, the sprite is still in its original scope, so won't be garbage collected

ricardoquesada replied:
Yes. probably we need to add that test as well. we still need to do a lot of tests.

pandamicro commented:
> Should we include this improvement in v3.10 or the future release?
> Now we have a separate v3.10 branch in the upstream.

@zilongshanren  It's not ready for v3.10, we need many test, I have following commits too, if all goes well, I think it can be alright to merge in v3.11

ricardoquesada committed:
restores autobindings

ricardoquesada commented:
@pandamicro @zilongshanren 

I think it should be part of v3.10, but it will be disabled. But advanced users can test if they want. I we can start telling them the benefits, receive feedback from them, etc.

@linshun thoughts?
